modules/util/path_test.go

```diff
@@ -138,13 +138,71 @@ func TestMisc_IsReadmeFileName(t *testing.T) {
 }
 
 func TestCleanPath(t *testing.T) {
-	cases := map[string]string{
-		"../../test": "test",
-		"/test":      "/test",
-		"/../test":   "/test",
+	cases := []struct {
+		elems    []string
+		expected string
+	}{
+		{[]string{}, ``},
+		{[]string{``}, ``},
+		{[]string{`..`}, `.`},
+		{[]string{`a`}, `a`},
+		{[]string{`/a/`}, `a`},
+		{[]string{`../a/`, `../b`, `c/..`, `d`}, `a/b/d`},
+		{[]string{`a\..\b`}, `a\..\b`},
+		{[]string{`a`, ``, `b`}, `a/b`},
+		{[]string{`a`, `..`, `b`}, `a/b`},
```

Comment:
I have thought about such case, it's caller's  duty, because caller want to trust user input as a "sub-part", that's why this case is here, the user said that "this sub-part will become empty". At the moment, I can't see an approach to stop such behavior with a clear definition -- unless we make a panic?

The most strict rule in my mind is: reject any "/../" parts, but I'm not sure most people would like this strict behavior.

ps: I haven't fully understood your question, in a real case. The case `{'lfs','repo/..', 'user/../path'}` is covered by the case in code IMO.

Could you help to elaborate about what would you like to do?

Reply:
Well… I think the case `{'lfs','repo/..', 'user/../path'}` would currently produce the result `lfs/path`, even if it is not exactly intended.
So, I simply wanted an extra test case to showcase that this behavior is known, even if its result might not be intended.